flask_app/blueprints/rest.py

```diff
@@ -128,6 +128,8 @@ def _get_iterator(self):
             args.session_id = request.view_args.get('session_id')
 
         if args.search:
+            if 'has_warnings' in args.search and args.session_id is None:
+                abort(requests.codes.forbidden)
```

Comment:
@vmalloc it will. any suggestions how to bypass it? I can check if args.search=='has_warnings' but if the search included another clause we get back to the DoS issue.

Reply:
I think the only option is to have some context in the search logic that can indicate we're "already narrowed down" to a single session...